src/xtgeo/xyz/_xyz.py

```diff
@@ -179,256 +152,193 @@ def from_file(self, pfile, fformat="xyz"):
+                )
+                continue
+
+            if cname not in self.dataframe:
+                if strict:
+                    raise ValueError(f"The column {cname} is not present.")
+                else:
+                    xtg.warnuser(f"Trying to delete {cname}, but it is not present.")
+            else:
+                self.dataframe.drop(cname, axis=1, inplace=True)
+
+    def operation_polygons(self, poly, value, opname="add", inside=True):
+        """A generic function for operations restricted to inside or outside polygon(s).
+
+        The operations are performed on the Z values, while the 'inside' or 'outside'
+        of polygons are purely based on X and Y values (typically X is East and Y in
+        North coordinates).
+
+        The operations are XYZ generic i.e. done on the points that defines the
+        Polygon or the point in Points, depending on the calling instance.
+
+        Possible ``opname`` strings:
+
+        * ``add``: add the value
+        * ``sub``: substract the value
+        * ``mul``: multiply the value
+        * ``div``: divide the value
+        * ``set``: replace current values with value
+        * ``eli``: eliminate; here value is not applied
 
-    def from_roxar(
-        self, project, name, category, stype="horizons", realisation=0, attributes=False
-    ):
-        """Load a points/polygons item from a Roxar RMS project.
+        Args:
+            poly (Polygons): A XTGeo Polygons instance
+            value(float): Value to add, subtract etc
+            opname (str): Name of operation... 'add', 'sub', etc
+            inside (bool): If True do operation inside polygons; else outside. Note
+                that boundary is treated as 'inside'
 
-        The import from the RMS project can be done either within the project
-        or outside the project.
+        Note:
+            This function works only intuitively when using one single polygon
+            in the ``poly`` instance. When having several polygons the
+            operation is done sequentially per polygon which may
+            lead to surprising results. For instance, using "add inside"
+            into two overlapping polygons, the addition will be doubled in the
+            overlapping part. Similarly using e.g. "eli, outside" will completely
+            remove all points of two non-overlapping polygons are given as input.
+        """
+        _xyz_oper.operation_polygons(self, poly, value, opname=opname, inside=inside)
 
-        Note that a shortform (for polygons) to::
+    def add_inside(self, poly, value):
+        """Add a value (scalar) to points inside polygons.
 
-          import xtgeo
-          mypoly = xtgeo.xyz.Polygons()
-          mypoly.from_roxar(project, 'TopAare', 'DepthPolys')
+        See notes under :meth:`operation_polygons`.
+        """
+        self.operation_polygons(poly, value, opname="add", inside=True)
```

Comment:
I have added tests for these "shiortforms" in my PR, see commit https://github.com/equinor/xtgeo/pull/598/commits/66cf139b89ede077822a627740b79ed389f824d8

Reply:
Added that test :)